official/resnet/resnet.py

```diff
@@ -164,10 +168,59 @@ def building_block(inputs, filters, training, projection_shortcut, strides,
     strides: The block's stride. If greater than 1, this block will ultimately
       downsample the input.
     data_format: The input format ('channels_last' or 'channels_first').
+    version: The ordering convention for convolution, batch normalization, and
+             activation.
 
   Returns:
     The output tensor of the block.
   """
+  if version == 1:
```

Comment:
I feel like if we are going to have a model class, we may as well use it to store state. Is it feasible to limit the use of version to the Model.__init__? For example, 

```
if version == 1:
  self.building_block_fn = _building_block_v1
elif version == 2:
  self.building_block_fn = _building_block_v2
```
and then just reference self.building_block_fn when the time comes?

Reply:
I moved the block_fn selection logic into the init. It's cleaner, but it will break things that inherit the resnet model. (Not sure if we make any interface promises on these classes.)